### PR TITLE
Adds turbolifts!

### PR DESCRIPTION
**Pull request by Kmc2000**
:cl: Kmc2000
add: Added turbolifts! Suffer not a crewman to have to climb up ladders no more.
/:cl:

https://i.gyazo.com/0c9beea615c934372584d0cb7ec38208.mp4

This is gonna need a lot of code review. I did some things for the sake of getting a prototype that are sorta messy.

Mapping these in is real simple, see documentation in the file.

They and their surrounding turfs are indestructible.. -- for now. It'd be mega immersion breaking if you could just break the walls down y'know?

ADDENDUM:
As soon as you step into a turbolift, it locks all the other turbolifts on that network so they can't be entered.

### CODE REVIEW

File: DS13/code/modules/turbolift/turbolift.dm [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+/turf/closed/wall/trek_smooth/indestructible/TerraformTurf(path, defer_change = FALSE, ignore_air = FALSE)
+	return
+
+/turf/closed/wall/trek_smooth/indestructible/acid_act(acidpwr, acid_volume, acid_id)
+	return 0
+
+/turf/closed/wall/trek_smooth/indestructible/attackby()
+	return FALSE //No deconning them
+
+/obj/structure/turbolift/process()
```
Comment (Jalleo) [inline DS13/code/modules/turbolift/turbolift.dm:83]: DENIED.

Use a system like the teleporter hub where it all connects up and one of them handles all this shit. Along with using a timer to know when they all unbolt.

**Review by Jalleo — DISMISSED** [review]
See comment for details. I wont review anything more than that for now.

File: DS13/code/modules/turbolift/turbolift.dm [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	if(is_controller)
```
Comment (spookydonut) [inline DS13/code/modules/turbolift/turbolift.dm]:
```suggestion
	if(!is_controller)
		return
```

File: DS13/code/modules/turbolift/turbolift.dm [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+	bolt_door()
+	in_use = TRUE
+	to_chat(user, floor_directory)
+	icon_state = "lift-off"
+	var/S = input(user,"Select a deck (max: [max_floor])") as num
+	if(S > max_floor || S <= 0 || S == floor)
+		in_use = FALSE
+		unbolt_door()
+		return
+	if(S)
```
Comment (spookydonut) [inline DS13/code/modules/turbolift/turbolift.dm]:
```suggestion
	if(!S)
		return
```

**Review by spookydonut — CHANGES_REQUESTED** [review]
erlireetern

**Review by spookydonut — APPROVED** [review]

**Pull request merged**